packages/orm/src/rules.ts

```diff
@@ -37,13 +37,13 @@ export type GenericError = { message: string };
 export type ValidationError = { entity: Entity } & GenericError;
 
 export class ValidationErrors extends Error {
-  public errors: ValidationError[];
+  public errors: Array<GenericError | ValidationError>;
   public readonly toJSON: () => string;
   constructor(errors: ValidationError[]);
   constructor(message: string);
```

Comment:
@chr1sjf0x I like that idea, but `ValidationError` currently requires an entity, and I don't think we know from the SQL error which entity specifically triggered the issue? 

I guess we are given this hint of what values triggered the error:

![image](https://github.com/joist-orm/joist-orm/assets/6401/ac2c7c12-3782-4079-a646-ab3236985ecc)

I.e. the `Key (publisher_id)=(1) already exists.` which I guess we could scan to find any entities have that value ... seems like it could get tricky to parse & match though.

So this is probably best...

cc @zgavin if any thoughts, but will approve to unblock & we can circle back for changes.